### PR TITLE
Required vs. allowed parameters in parameter table

### LINKED ISSUE #219: All parameters need to be in parameters sheet
At the moment all parameters in the SBML model need to be present in the parameters sheet (also the fixed parameters that are not subject to optimization). Might be more convenient to make it optional to define fixed parameters there and otherwise use the values from the SBML model.

Any opinions?

### PR DESCRIPTION
**Pull request by dweindl**
Closes #219

- Update required vs. allowed parameters in docs
- Correctly check for required vs. allowed parameters

### CODE REVIEW

File: petab/parameters.py [hunk 1]
```diff
@@ -117,13 +130,68 @@ def get_required_parameters_for_parameter_table(
...
+
+    '''
+    - grab all model parameters
+    - grab all parameters from measurement table
+    - grab all parametric overrides from condition table
+    - remove parameters for which condition table columns exist
+    - remove observables assigment targets
+    - remove sigma assignment targets
+    - remove placeholder parameters (only partial overrides are not supported)
+    '''
```
Comment (dweindl) [inline petab/parameters.py]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=4851634)

Comment (dweindl) [inline petab/parameters.py]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=4851634)

File: petab/parameters.py [hunk 2]
```diff
@@ -117,13 +130,67 @@ def get_required_parameters_for_parameter_table(
...
+        sbml_model: PEtab SBML model
+        condition_df: PEtab condition table
+        measurement_df: PEtab measurement table
+
+    Returns:
+        Set of parameter IDs which PEtab allows to be present in the
+        parameter table.
+    """
+
+    # - grab all model parameters
```
Comment (yannikschaelte) [inline petab/parameters.py]: alternatively in docstring?

Comment (dweindl) [reply petab/parameters.py]: That's rather implementation-specific, would keep it separate.

**Pull request merged**